src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -1,88 +1,69 @@
+using MarkdownLog;
+using Microsoft.Xunit.Performance.Api.Table;
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
 using System.Linq;
-using System.Reflection;
-using Microsoft.Xunit.Performance;
-using Microsoft.Xunit.Performance.Api.Table;
+using System.Xml.Linq;
+using static Microsoft.Xunit.Performance.Api.XunitPerformanceLogger;
 
 namespace Microsoft.Xunit.Performance.Api
 {
     public sealed class XunitPerformanceHarness : IDisposable
     {
-        private string[] _args;
-
         public XunitPerformanceHarness(string[] args)
         {
             _args = args;
+            _disposed = false;
 
             // Set the run id.
-            Configuration.RunId = DateTimeOffset.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss");
+            Configuration.RunId = $"{DateTimeOffset.UtcNow.ToString("yyyyMMddHHmmss")}";
 
             // Set the file log path.
             // TODO: Conditionally set this based on whether we want a csv file written.
             Configuration.FileLogPath = Configuration.RunId + ".csv";
         }
 
-        void IDisposable.Dispose()
+        public BenchmarkConfiguration Configuration
         {
-            // Close the log when all test cases have completed execution.
-            // TODO: This is a hack because we haven't found a way to close the file from within xunit.
-            BenchmarkEventSource.Log.Close();
-
-            // Process the results now that we know we're done executing tests.
-            ProcessResults();
+            get { return BenchmarkConfiguration.Instance; }
         }
 
-        public BenchmarkConfiguration Configuration
+        public int RunBenchmarks(string assemblyPath)
         {
-            get { return BenchmarkConfiguration.Instance; }
+            Validate(assemblyPath);
+            int errCode = 0;
+            ETWProfiler.Profile(assemblyPath, Configuration.RunId, () => { errCode = XunitRunner.Run(assemblyPath); });
+            return errCode;
         }
 
-        public void RunBenchmarks(string assemblyPath)
+        private static void Validate(string assemblyPath)
```

Comment:
Yes, it could be on the RunBenchmarks method. I just like to have the validation code in a separate method. I think it's cleaner and moving forward we will be getting more parameters that need validation (e.g. args[]).

---
In reply to: [100351798](https://github.com/Microsoft/xunit-performance/pull/183#discussion_r100351798) [](ancestors = 100351798)